pandas/tseries/period.py

```diff
@@ -281,6 +281,16 @@ def __contains__(self, key):
             return False
         return key.ordinal in self._engine
 
+    def __array_wrap__(self, result, context=None):
```

Comment:
PeriodIndex + integer ndarray should not be expected to work -- the array
does not have the right type!

On Mon, Aug 17, 2015 at 3:35 PM, Sinhrks <notifications@github.com> wrote:

> In pandas/tseries/period.py
> <https://github.com/pydata/pandas/pull/10638#discussion_r37245206>:
>
> > @@ -281,6 +281,16 @@ def __contains__(self, key):
> >              return False
> >          return key.ordinal in self._engine
> >
> > +    def __array_wrap__(self, result, context=None):
>
> Raising TypeError in __array_wrap__ affects to arithmetic using np.array.
> Currently, it works:
>
> pd.PeriodIndex(['2011', '2012'], freq='A') + np.array([1, 2])
> # PeriodIndex(['2012', '2014'], dtype='int64', freq='A-DEC')
>
> So it can't be prohibited?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10638/files#r37245206>.
>